tfhe/src/integer/server_key/radix_parallel/tests_signed.rs

```diff
@@ -1474,6 +1518,342 @@ fn integer_signed_default_absolute_value(param: impl Into<PBSParameters>) {
     }
 }
 
+fn integer_signed_default_left_shift<P>(param: P)
+where
+    P: Into<PBSParameters>,
+{
+    let (cks, mut sks) = KEY_CACHE.get_from_params(param);
+    let cks = RadixClientKey::from((cks, NB_CTXT));
+
+    sks.set_deterministic_pbs_execution(true);
+
+    let mut rng = rand::thread_rng();
+
+    let modulus = (cks.parameters().message_modulus().0.pow(NB_CTXT as u32) / 2) as i64;
+
+    assert!(modulus > 0);
+    assert!((modulus as u64).is_power_of_two());
+    let nb_bits = modulus.ilog2() + 1; // We are using signed numbers
```

Comment:
how does that change the nb_bits and increases it ?

Reply:
Above, we computed the modulus as `let modulus = (cks.parameters().message_modulus().0.pow(NB_CTXT as u32) / 2) as i64;` so the unsigned modulus divided by two. So two have the number of bits we have to add that bit

Reply:
indeed